package/sbp_anpp_bridge/src/sbp_anpp_bridge.c

```diff
@@ -0,0 +1,243 @@
+/*
+ * Copyright (C) 2017 Swift Navigation Inc.
+ * Contact: Swift Navigation <dev@swiftnav.com>
+ *
+ * This source is subject to the license found in the file 'LICENSE' which must
+ * be be distributed together with this source. All other rights reserved.
+ *
+ * THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+ * EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+ */
+
+#include <assert.h>
+#include <getopt.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <time.h>
+
+#include <libpiksi/logging.h>
+#include <libpiksi/settings_client.h>
+#include <libpiksi/util.h>
+
+#include <libsbp/sbp.h>
+#include <libsbp/vehicle.h>
+
+#include <libsettings/settings.h>
+
+#include "an_packet_protocol.h"
+#include "sbp.h"
+#include "obdii_odometer_packets.h"
+
+#define PROGRAM_NAME "sbp_anpp_bridge"
+
+#define ANPP_SUB_ENDPOINT "ipc:///var/run/sockets/anpp_internal.pub" /* ANPP Internal Out */
+#define ANPP_SUB_METRICS "anpp/sub"
+
+#define ANPP_PUB_ENDPOINT "ipc:///var/run/sockets/anpp_internal.sub" /* ANPP Internal In */
+#define ANPP_PUB_METRICS "anpp/pub"
+
+#define M_TO_MM(M) ((M)*1000)
+
+#define SBP_MSG_ODOMETRY_FLAGS_NOTIME (0x00u)
+#define SBP_MSG_ODOMETRY_FLAGS_GPSTIME (0x01u)
+#define SBP_MSG_ODOMETRY_FLAGS_SYSTIME (0x02u)
+
+#define SBP_MSG_ODOMETRY_FLAGS_SRC0 (0x00u)
+#define SBP_MSG_ODOMETRY_FLAGS_SRC1 (0x04u)
+#define SBP_MSG_ODOMETRY_FLAGS_SRC2 (0x08u)
+#define SBP_MSG_ODOMETRY_FLAGS_SRC3 (0x0Cu)
+
+bool anpp_debug = false;
+
+pk_endpoint_t *anpp_pub = NULL;
+
+static u32 get_time_ms(void)
+{
+  struct timespec now;
+  clock_gettime(CLOCK_MONOTONIC, &now);
+  return (now.tv_sec * 1000 + now.tv_nsec / 1000000);
+}
+
+static void convert_anpp_odometer_to_sbp(odometer_packet_t *an_odo_packet, msg_odometry_t *odo_msg)
+{
+  odo_msg->tow = (u32)get_time_ms();
```

Comment:
If we use linux time here, we should probably plan to include a fudge factor for the 30 - 60 ms offset between actual time of week and the linux system time of week inside the pose_daemon wrapper since we don't have precision time of week in linux.  We should be able to tune this fudge factor via post-processing I think.